From 2ae6e5970d1edf322798c760375e54249cee4633 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:04:29 -0400
Subject: [acpi] Drop unnecessary label from acpi_bind_one()

Message-id: <20140828030429.2117.97454.stgit@gir.stowe>
Patchwork-id: 89297
O-Subject: [RHEL7.1 PATCH 09/30] ACPI: Drop unnecessary label from acpi_bind_one()
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: 3342c753bdeb29ec29d721c7ce38d283cc969174

commit 3342c753bdeb29ec29d721c7ce38d283cc969174
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Aug 8 16:19:10 2013 +0200

    ACPI: Drop unnecessary label from acpi_bind_one()

    The out_free label in acpi_bind_one() is only jumped to from one
    place, so in fact it is not necessary, because the code below it
    can be moved to that place directly.  Move that code and drop the
    label.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index dcba319..f3ead0c 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -225,11 +225,15 @@ int acpi_bind_one(struct device *dev, acpi_handle handle)
  list_for_each_entry(pn, &acpi_dev->physical_node_list, node) {
   /* Sanity check. */
   if (pn->dev == dev) {
+   mutex_unlock(&acpi_dev->physical_node_lock);
+
    dev_warn(dev, "Already associated with ACPI node\n");
-   if (ACPI_HANDLE(dev) == handle)
-    retval = 0;
+   kfree(physical_node);
+   if (ACPI_HANDLE(dev) != handle)
+    goto err;
 
-   goto out_free;
+   put_device(dev);
+   return 0;
   }
   if (pn->node_id == node_id) {
    physnode_list = &pn->node;
@@ -262,15 +266,6 @@ int acpi_bind_one(struct device *dev, acpi_handle handle)
  ACPI_HANDLE_SET(dev, NULL);
  put_device(dev);
  return retval;
-
- out_free:
- mutex_unlock(&acpi_dev->physical_node_lock);
- kfree(physical_node);
- if (retval)
-  goto err;
-
- put_device(dev);
- return 0;
 }
 EXPORT_SYMBOL_GPL(acpi_bind_one);
 
-- 
1.7.1